### PR TITLE
feat: add progress update card component with initial test cases

### PR DESCRIPTION
**Pull request by satyam73**
### Issue:
- #1053 
### Description:

Please refer issue ticket for full user story and acceptance criteria.

### Dev Tested:
- [x] Yes

### Images/video of the change:

image
> Mobile devices

image
>Tablets 

image
> Desktops

image
> With tooltip visible

### Follow-up Issues (if any)
n/a

### Tests Coverage
image
image

### CODE REVIEW

File: .env.development [hunk 1]
```diff
@@ -1,4 +1,4 @@
-NEXT_PUBLIC_BASE_URL='https://staging-api.realdevsquad.com'
+NEXT_PUBLIC_BASE_URL='https://api.realdevsquad.com'
```
Comment (vinit717) [inline .env.development:1]: Why you change this for development

File: src/utils/common.ts
Comment (vinit717) [inline src/utils/common.ts:1]: where is the test for this?

File: src/components/taskDetails/ProgressDetails.tsx
[the remark this answers is not in the document]
Comment (satyam73) [reply src/components/taskDetails/ProgressDetails.tsx:1]: This component was there before I have only modified some part of it which is in dev flag

File: src/components/taskDetails/ProgressUpdateCard/LatestProgressUpdateCardPresentation.tsx
[the remark this answers is not in the document]
Comment (satyam73) [reply src/components/taskDetails/ProgressUpdateCard/LatestProgressUpdateCardPresentation.tsx:1]: check ProgressUpdateCard tests

File: src/components/taskDetails/ProgressUpdateCard/ProgressUpdateCardPresentation.tsx
[the remark this answers is not in the document]
Comment (satyam73) [reply src/components/taskDetails/ProgressUpdateCard/ProgressUpdateCardPresentation.tsx:1]: check ProgressUpdateCard tests

**Comment by satyam73** [comment]
> image Why this is being in another block

Fixed 
image

File: src/components/ProgressCard/index.tsx [hunk 2]
```diff
@@ -1,22 +1,26 @@
...
 
 type SortedProgressType = { data: ProgressDetailsData[]; order: number };
 
 type Props = { taskProgress: ProgressDetailsData[] };
 
 export default function ProgressCard({ taskProgress }: Props) {
     const [sortedProgress, setSortedProgress] = useState<SortedProgressType>();
-
+    const router = useRouter();
+    const isDev = router.query.dev === 'true';
```
Comment (vinayak-trivedi) [inline src/components/ProgressCard/index.tsx:16]: if we using router only for this, can we not change the code like this

```js
const isDev = useRouter().query.dev === 'true';
```

File: src/components/ProgressCard/index.tsx [hunk 3]
```diff
@@ -39,6 +43,27 @@ export default function ProgressCard({ taskProgress }: Props) {
...
+            sortedProgress.order === 0 &&
+            isDev
+        ) {
+            return (
+                <LatestProgressUpdateCard key={progress.id} data={progress} />
+            );
+        }
+
+        return <ProgressDetails key={progress.id} data={progress} />;
+    });
```
Comment (vinayak-trivedi) [inline src/components/ProgressCard/index.tsx:65]: can we fix the indentation here?

**Review by vinayak-trivedi — CHANGES_REQUESTED** [review]

File: src/components/ProgressCard/index.tsx [hunk 4]
```diff
@@ -39,6 +43,27 @@ export default function ProgressCard({ taskProgress }: Props) {
...
+        if (idx === 0 && sortedProgress?.order === 1 && isDev) {
+            return (
+                <LatestProgressUpdateCard key={progress.id} data={progress} />
+            );
+        }
+
+        if (
+            sortedProgressLength &&
+            idx === sortedProgressLength - 1 &&
+            sortedProgress.order === 0 &&
```
Comment (ajoykumardas12) [inline src/components/ProgressCard/index.tsx:56]: Can optional chain here, `sortedProgress?.order`

**Comment by ajoykumardas12** [comment]
Also, from the UI, it's not clear what `17 days ago` (first update card) implies.

**Comment by satyam73** [comment]
> Also, from the UI, it's not clear what `17 days ago` (first update card) implies.

It is intended to be like that if you will look into the section in which the card comes under it will make clear sense of this data.

File: src/components/ProgressCard/index.tsx (hunk 3 above)
Comment (satyam73) [reply src/components/ProgressCard/index.tsx]: it is happening after I run yarn prettier-fix I think it's project's configuration

**Review by ajoykumardas12 — APPROVED** [review]

**Review by vinit717 — APPROVED** [review]

**Pull request merged**